pkg/provider/ip/provider.go

```diff
@@ -192,7 +192,7 @@ func (p *ipv4Provider) DeInitResource(instance ec2.EC2Instance) error {
 func (p *ipv4Provider) UpdateResourceCapacity(instance ec2.EC2Instance) error {
 	resourceProviderAndPool, isPresent := p.getInstanceProviderAndPool(instance.Name())
 	if !isPresent {
-		p.log.Error(nil, "cannot find the instance provider and pool form the cache", "node-name", instance.Name())
+		p.log.Error(utils.ErrNotFound, "cannot find the instance provider and pool form the cache", "node-name", instance.Name())
 		return nil
```

Comment:
My question pertains to the next line, where I should return `nil` or the actual error `utils.ErrNotFound` so that the caller can process it.